### PR TITLE
[merged] Add a "deploy" command and remove arg from "upgrade"

### PR DESCRIPTION
**Pull request by mbarnes**
Command-line wrapper for the "deploy" operation.

Depends on https://github.com/projectatomic/commissaire/pull/135

### CODE REVIEW

**Comment by ashcrow** [comment]
Looking now.

File: src/commctl/cli.py [hunk 1]
```diff
@@ -277,6 +277,32 @@ def host_delete(self, address, **kwargs):
...
+    def cluster_deploy_status(self, name, **kwargs):
+        """
+        Attempts to get the status of an ongoing tree image deployment.
+
+        :param name: The name of the cluster
+        :type name: str
+        :param kwargs: Any other keyword arguments
+        :type kwargs: dict
+        """
+        path = '/api/v0/cluster/{0}/deploy'.format(name)
```
Comment (ashcrow) [inline src/commctl/cli.py:289]: Totally not something that needs to be done here but it hit me when reading this that we should probably have something like:

``` python
_API_PREFIX = '/api/{version}'
_DEFAULT_VERSION = 0

def make_endpoint(version=_DEFAULT_VERSION, *path):
    endpoint = _API_PREFIX.format(version)
    for item in path:
        endpoint += '/{0}.format(item)
    return endpoint
```

File: src/commctl/cli.py [hunk 2]
```diff
@@ -544,6 +570,23 @@ def add_cluster_commands(argument_parser):
...
+
+    object_parser = subject_subparser.add_parser('deploy')
+    object_subparser = object_parser.add_subparsers(dest='subcommand')
+    object_subparser.required = True
+
+    # Sub-command: cluster deploy start
+    verb_parser = object_subparser.add_parser('start')
+    verb_parser.required = True
+    verb_parser.add_argument('name', help='Name of the cluster')
+    verb_parser.add_argument('version', help='Version to deploy')
```
Comment (ashcrow) [inline src/commctl/cli.py:583]: Would it make sense to require version for deploy?

**Comment by ashcrow** [comment]
One question. Overall looks good.

File: src/commctl/cli.py (hunk 2 above)
Comment (mbarnes) [inline src/commctl/cli.py:583]: Not sure I understand.  The `version` argument _is_ required.  Or are you talking package dependencies?

Comment (ashcrow) [inline src/commctl/cli.py:583]: No I meant the argument. I guess I'm used to seeing `required=True` so much when I don't see it I make an assumption. No change required. Looks good.

**Comment by ashcrow** [comment]
@rh-atomic-bot try

**Comment by ashcrow** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**